lib/src/util/posix/dynamic_library.cc

```diff
@@ -41,7 +41,11 @@ namespace facter { namespace util {
         close();
 
         // Don't actually perform a load to determine if it is already loaded
+#ifdef _AIX
+        _handle = dlopen(name.c_str(), RTLD_LAZY);
```

Comment:
+1 on a discrete commit for this workaround. But if it's accompanied by an explanatory comment (something about "Workaround pending resolution of FACT-891"), this seems reasonable enough. 

And then if we end up living with some of these `ifdef`'s long-term for whatever reason, it would be nice to segregate them in some platform driven headers.